### ISSUE TITLE
Validation metrics exceeding training metrics

### ISSUE THREAD

**Issue by aghand0ur** [qubvel/segmentation_models#294]
HI @qubvel, Hi all,

I am training a UNet classifier for the OpenCitiesAI challenge (1 building class and ResNext50 backbone). @hasan-nn

At the end of the first epoch, I got the following metrics where the validation metrics are much better than the test metrics:

`23366/23366 [==============================] - 15937s 682ms/step - loss: 0.2726 - iou_score: 0.5777 - f1-score: 0.7036 - val_loss: 0.0748 - val_iou_score: 0.7403 - val_f1-score: 0.7849 -
`

This issue is kind of related to the issue below:
https://github.com/qubvel/segmentation_models/issues/285

Please advise.

Ali

**Comment by qubvel** [comment]
Hi, make sure you calculate jaccard micro, not macro metric. Read rules more careful.

**Comment by aghand0ur** [comment]
> Hi, make sure you calculate jaccard micro, not macro metric. Read rules more careful.

Hi Pavel @qubvel,
Thanks for your prompt reply.
For single class detection, I guess micro and macro Jaccard turn out to be the same as shown below:

` 2705/23366 [==>...........................] - ETA: 3:15:54 - loss: 0.2258 - iou_score: **0.6489** - f1-score: 0.7620 - hard_jacard_coef_building: **0.6489** - Jaccard_micro_building **0.6489**`
 
Nevertheless, my initial question was related to validation results. At the end of epoch, I notice that val_jaccard is way higher than test_jaccard. @hasan-nn

Ali

**Comment by qubvel** [comment]
```
'micro':
Calculate metrics globally by counting the total true positives, false negatives and false positives.

'macro':
Calculate metrics for each label, and find their unweighted mean. This does not take label imbalance into account.

'samples':
Calculate metrics for each instance, and find their average (only meaningful for multilabel classification).

```
yeah, I mean make sure you calculate total intersection and total union and estimate the score (not for each image/batch and then average)

Cannot see any comparison of *test* and *validation* metrics in your question

**Comment by hasan-nn** [comment]
Between Train * and Validation * @qubvel

**Comment by aghand0ur** [comment]
Yes, thanks @hasan-nn. That was a typo. 

I meant between training metrics and validation metrics. @qubvel

**Comment by aghand0ur** [comment]
As pointed by @hasan-nn, this issue might be caused by the fact that the batch size for validation was different than the the batch size for training.

**Comment by aghand0ur** [comment]
Hi Pavel @qubvel,

Hope you are staying safe.

Congratulations for ranking 1st in OpenCitiesAI challenge. Are you planning to publish your solution soon?

Ali

**Comment by qubvel** [comment]
Thanks, my solution is based on segmentation_models.pytorch. I don't know if I can publish it now, hope to publish it as soon as solution will be validated by organizers.

**Comment by aghand0ur** [comment]
Please note that 2nd is already posted on github and 3rd solution is posted on drivendata and github.

**Comment by qubvel** [comment]
Where did your find second rank solution, can you share the link?

**Comment by hasan-nn** [comment]
@qubvel
this is the 2nd position solution repository : 
https://github.com/kbrodt/building-segmentation-disaster-resilience
He also used your pytorch library
